docs/section22/Rule22-25.md

```diff
@@ -0,0 +1,53 @@
+
+# CHW&CW - Rule 22-25  
+
+**Schema Version:** 0.0.10 **Mandatory Rule:** True  
+**Rule ID:** 22-25  
+**Rule Description:** For baseline chilled-water systems served by chiller(s), the primary pump shall be modeled as constant volume.  
+**Rule Assertion:** B-RMR = expected value  
+**Appendix G Section:** Section 22 CHW&CW Loop  
+**90.1 Section Reference:** Section G3.1.3.10 Chilled-water pumps (Systems 7, 8, 11, 12, and 13)  
+**Data Lookup:** None  
+**Evaluation Context:** Building  
+**Applicability Checks:**  
+
+1. B-RMR is modeled with at least one air-side system that is Type-7, 8, 11.1, 11.2, 12, 13, 7b, 8b, 11b, 12b, 13b.
+2. B-RMR is modeled with primary-secondary configuration.
+
+**Function Calls:**  
+
+1. get_baseline_system_types()
+2. get_primary_secondary_loops()
+3. get_component_by_id()
+
+**Applicability Checks:**  
+
+- **Check 1:** Get B-RMR system types: `baseline_hvac_system_dict = get_baseline_system_types(B-RMR)`
+
+  - Check if B-RMR is modeled with at least one air-side system that is Type-7, 8, 11.1, 11.2, 12, 13, 7b, 8b, 11b, 12b, 13b, i.e. with air-side system served by chiller(s), continue to the next applicability check: `if any(sys_type in baseline_hvac_system_dict.keys() for sys_type in ["SYS-7", "SYS-8", "SYS-11.1", "SYS-11.2", "SYS-12", "SYS-13", "SYS-7B", "SYS-8B", "SYS-11B", "SYS-12B", "SYS-13B"]): NEXT_APPLICABILITY_CHECK`
+
+  - Else, rule is not applicable to B-RMR: `else: RULE_NOT_APPLICABLE`
+
+- **Check 2:** Get primary and secondary loops for B-RMR: `primary_secondary_loop_dictionary = get_primary_secondary_loops(B_RMR, "COOLING")`
+
+  - Check if B-RMR is modeled with primary secondary configuration, continue to rule logic: `if LEN(primary_secondary_loop_dictionary) != 0: CHECK_RULE_LOGIC`
+
+  - Else, rule is not applicable to B-RMR: `else: RULE_NOT_APPLICABLE`
+
+## Rule Logic:
+
+- For each pump in B_RMR: `for pump_b in B_RMR.RulesetModelInstance.pumps:`
```

Comment:
Why not just get pumps that reference primary loops returned by get_primary_secondary_loops(B_RMR, "COOLING")`?

Reply:
As we are not able to get Pump directly from a FluidLoop (no pump data element under fluidloop), Line 39 through 41 save all loops and their associated pumps for Line 45 to only get pumps associated with primary loops so that we only need to loop through all pumps once.